VMEncryption/main/check_util.py

```diff
@@ -273,14 +273,31 @@ def is_supported_os(self, public_settings, DistroPatcher, encryption_status):
                             return
             raise Exception('Distro {0} {1} is not supported for OS encryption'.format(distro_name, distro_version))
 
-    def precheck_for_fatal_failures(self, public_settings, encryption_status, DistroPatcher):
+    def validate_volume_type_for_enable(self, public_settings, existing_volume_type):
+        encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
+        if not encryption_operation in [CommonVariables.EnableEncryption, CommonVariables.EnableEncryptionFormat, CommonVariables.EnableEncryptionFormatAll]:
+            self.logger.log("Current operation is not an enable. Skipping volume type validation.")
+            return
+        if not existing_volume_type:
+            self.logger.log('Existing volume type not found. First enable.')
+            return
+        volume_type = public_settings.get(CommonVariables.VolumeTypeKey)
+        if volume_type.lower() == existing_volume_type.lower():
+            self.logger.log('No change in data type.')
```

Comment:
Expected: 'No change in **volume** type'

Reply:
Done.